src/tv2-common/helpers/dsk.ts

```diff
@@ -74,6 +74,23 @@ export function getDskOnAirTimelineObjects(
 						}
 					})
 			  ]
+			: []),
+		...(dskRole === DskRole.FULLGFX && context.uniformConfig.switcherLLayers.fullUskMixEffect
```

Comment:
A small thing you could would be to extract it to a method so it would look something like this
```
export function getDskOnAirTimelineObjects(...) {
  ...
  return [
    context.videoSwithcer.getDskTimelineObject(...),
    ...getJingleUskMixEffectTimelineObject(...),
    ...getFullGfxUskMixEffectTimelineObject(...)
  ]
}

function getFullGfxUskMixEffectTimelineObject(...) {
  if (dskRole !== DskRole.FULLGFX || !context.uniformConfig.switcherLLayers.fullUskMixEffect) {
    return []
  }
  return [
    context.videoSwitcher.getMixEffectTimelineObject(...)
  ]
}
```

But maybe that's not what you are unhappy with this about?

Reply:
That would be an improvement as far as function length is concerned, but I'm still not convinced it logically makes the most sense. We do have the three functions: `findDskFullGfx`, `findDskOverlayGfx`, `findDskJingle` that hide the `DskRole` away, so maybe `getDskOnAirTimelineObjects` could be turned into three separate functions that have the role in the name as well, but then we end up with more and more exported loose functions...